ui/src/utils/url.ts

```diff
@@ -1 +1,9 @@
-export const urlWithProtocol = (url: string) => `https://${url.replace('https://', '')}`;
+export const urlWithProtocol = (image: string): string | undefined => {
+  if (image.startsWith('quay.io') || image.startsWith('docker.io') || image.startsWith('ghcr.io')) {
+    return `https://${image}`;
+  } else if (image.includes('/')) {
+    return `https://hub.docker.com/r/${image}`;
+  } else {
+    return `https://hub.docker.com/_/${image}`;
+  }
+};
```

Comment:
Some of this looks iffy to me.

I know an image like `docker.io/krancour/kargo`, for instance, cannot be accessed at http://docker.io/krancour/kargo. That's a 404.

It _does_ work for ghcr.io.

```
} else if (image.includes('/')) {
    return `https://hub.docker.com/r/${image}`;
  } else {
    return `https://hub.docker.com/_/${image}`;
  }
 ```
 
 This part also looks iffy because there would be many images that would contain a forward slash that aren't on Docker Hub.

Reply:
For context, I think by nature this feature will have to be a bit iffy, or at least will be brittle and break if your image doesn't hold up to one of the 4 patterns we're accounting for (quay, ghcr, docker official and docker hub)

You're right that `docker.io/krancour/kargo` gives a 404, but wouldn't this image be referenced as just `krancour/kargo`? If it's common to reference it with the `docker.io` prefix I'll change the logic but I didn't think it was. Continuing with the `krancour/kargo` example, this would trip the second `if` condition and redirect properly to Docker Hub.  

You can think of the forward slash as _only_ differentiating between a Docker Hub image or Docker Official image; if we get past the first `if` condition, the image isn't from one of the registries we committed to recognizing for this feature, so we assume it is from Docker itself and just determine if it's official or not. You're right that any image in any registry might have a forward slash in its name, but I don't think it's possible to recognize every possible repository.

I'd like to introduce a new label (something like `kargo.akuity.io/repository`) which would allow someone to bypass this logic. There are some hurdles to overcome to make that work but it would allow customization for unique image repos that we couldn't support natively.